lib/xmalloc.c

```diff
@@ -340,6 +340,7 @@ xrealloc_impl(void *ptr, size_t new_size, const char *file, int line,
   new_ptr = realloc(ptr, new_size);
   if (new_ptr != NULL)
```

Comment:
Next, we check if we were successful.

As a side note, we could save ourselves a bit of work if we checked not only that `new_ptr` is not `NULL` but also that it is not equal to `ptr`, which can happen, especially if `new_size` is either smaller than the original size, or only very slightly larger.